lib/practice_exercises.rb

```diff
@@ -1,13 +1,45 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string_array = string.split("")
+  my_char_array = string_array.select do |char|
+    /[a-zA-Z]/.match char
+  end
+
+  first_index = 0
+  last_index = my_char_array.length - 1
+  while first_index < last_index
+    if my_char_array[first_index].upcase == my_char_array[last_index].upcase
+      first_index += 1
+      last_index -= 1
+    else
+      return false
+    end
+  end
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: O(n)
+
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍